### PR TITLE
Block by tracking IP addresses

### PR DESCRIPTION
**Pull request by lallousx86**
-  Block by IP addresses
- Cache IP resolution
- Clear blocking rules option

### CODE REVIEW

**Comment by 10se1ucgo** [comment]
Not really liking the weird "constants" section, also a ton of weird formatting issues.
Plus seems like a lot of code for just blocking IPs. Looks pretty hard to read.

**Comment by lallousx86** [comment]
Don't worry. I plan to port this tool to C++, it will be a lightweight tool that require no dependencies and just around 200kb at most download.

**Comment by 10se1ucgo** [comment]
eh? i'm not worried, it's just the coding style is awful, I've never seen anyone ever write code in that format. I'll write my own IP blocking later.

**Comment by lallousx86** [comment]
"it's just the coding style is awful. I've never seen anyone ever write code in that format. I'll write my own IP blocking later" I don't know what you're complaining about and what is "awful". But thanks for the feedback.

The formatting? that's just my editor being misconfigured.

L17-L25: these are regular constants. I could put them outside in the module scope or in the class. That's usual. I have seen it a lot.

L435-L442: that's something I overlooked. A copy/paste bug. The "and 0 self.servicebox.IsChecked()" is not needed.

So basically, you check if that checkbox is set once and then inside that check you check the radio button value. It is cleaner then checking "self.servicebox.IsChecked()" twice.

* Tabs and spaces? I had a fresh PC and my editor (EditPlus) was not yet configured, so that explains this mix and match between spaces and tabs.

**Comment by lallousx86** [comment]
It was really messy (the tabs and spaces). I fixed it.

Why do you use class variables as local variables by the way? (self.MSHosts2, etc...).
You can just use local variables all over the place instead.

**Comment by 10se1ucgo** [comment]
Yeah I changed it in the rewrite. I'll push the code to the 2.0 branch in the morning.
